frontend/src/stack/create/create-by-Json.html

```diff
@@ -6,7 +6,7 @@
         <div class="form-control">
             <input id="json-create-Namespace" type="text" data-ng-model="stackCreateByJsonCtrl.form.Namespace"
                    data-required="required" name="Name"
-                   data-ng-pattern="/^[A-Za-z0-9]+$/" focus-me>
+                   data-ng-pattern="/^\w[\w_.-]+\w$/" focus-me>
```

Comment:
Here the **namespace** is **Stack Name** instead, not **service name** .

At the same time, the service name format is StackNamespace_ServiceName deployed by crane. We have to avoid the charactor *_* in stack namespace pattern , in case of the split confuse.

So make data-ng-pattern="/^\w[\w.-]?\w$/" , IMO.